client/src/test/java/com/linecorp/decaton/client/internal/DecatonClientImplTest.java

```diff
@@ -155,6 +156,37 @@ public void testWithEmptyTaskMetaDataSetExternally() {
         assertNotNull(record.value().getMetadata().getSourceInstanceId());
     }
 
+    @Test
+    public void testSpecifyingPartition() {
+        doReturn(1234L).when(timestampSupplier).get();
+
+        client.put("key", HelloTask.getDefaultInstance(),
+                   TaskMetadata.builder()
+                               .timestamp(5678L)
+                               .scheduledTime(6912L)
+                               .build(), 4);
+
+        verify(producer, times(1)).send(captor.capture(), any(Callback.class));
+        ProducerRecord<byte[], DecatonTaskRequest> record = captor.getValue();
+        assertNotNull(record.partition());
+        assertEquals(4, record.partition().intValue());
+        assertNull(record.timestamp());
+        assertEquals(5678L, record.value().getMetadata().getTimestampMillis());
+        assertEquals(6912L, record.value().getMetadata().getScheduledTimeMillis());
+    }
+
+    @Test
+    public void testSpecifyingPartitionWithoutMetadata() {
+        doReturn(1234L).when(timestampSupplier).get();
+
+        client.put("key", HelloTask.getDefaultInstance(), TaskMetadata.builder().build(), 4);
```

Comment:
Isn't this test supposed to pass `null` as the taskmetadata argument? (rather than empty task metadata)